examples/go/dataSubmitWatch/main.go

```diff
@@ -0,0 +1,163 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"avail-gsrpc-examples/internal/config"
+	"avail-gsrpc-examples/internal/extrinsics"
+
+	gsrpc "github.com/centrifuge/go-substrate-rpc-client/v4"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/signature"
+	"github.com/centrifuge/go-substrate-rpc-client/v4/types"
+)
+
+// The following example shows how submit data blob and track transaction status
+func main() {
+	var configJSON string
+	var config config.Config
+	flag.StringVar(&configJSON, "config", "", "config json file")
+	flag.Parse()
+
+	if configJSON == "" {
+		log.Println("No config file provided. Exiting...")
+		os.Exit(0)
+	}
+
+	err := config.GetConfig(configJSON)
+	if err != nil {
+		panic(fmt.Sprintf("cannot get config:%v", err))
+	}
+
+	api, err := gsrpc.NewSubstrateAPI(config.ApiURL)
+	if err != nil {
+		panic(fmt.Sprintf("cannot create api:%v", err))
+	}
+
+	meta, err := api.RPC.State.GetMetadataLatest()
+	if err != nil {
+		panic(fmt.Sprintf("cannot get metadata:%v", err))
+	}
+
+	// Set data and appID according to need
+	size := 100
+	if config.Size > 0 {
+		size = config.Size
+	}
+
+	subData, _ := extrinsics.RandToken(size)
+	fmt.Println("Submitting data ...")
+	appID := 0
```

Comment:
I thought app extrinsics couldn't be submittted with appID=0, which is for every other type of extrinsic, and only sudo can do that, but I may be wrong.